tests/js/server/aql/aql-vpack-index-rearming.js

```diff
@@ -0,0 +1,230 @@
+
+      let nodes = AQL_EXPLAIN(q, null, opts).plan.nodes;
+      let indexNodes = nodes.filter((node) => node.type === 'IndexNode');
+      assertEqual(1, indexNodes.length);
+      assertEqual(1, indexNodes[0].indexes.length);
+      assertEqual("UnitTestsIndex", indexNodes[0].indexes[0].name);
+
+      let qr = db._query(q, null, opts);
+      let stats = qr.getExtra().stats;
+      assertEqual(1, stats.cursorsCreated);
+      assertEqual(n - 2, stats.cursorsRearmed);
+      let results = qr.toArray();
+      assertEqual(n - 1, results.length);
+
+      for (let i = 0; i < n - 1; ++i) {
+        let doc = results[i];
+        assertEqual(prefix + String(i).padStart(5, "0"), doc.value1);
+        assertEqual(i, doc.value2);
+      }
+    },
+    
+    testLookupByRangeOnFirstAttribute : function () {
+      const q = `FOR i IN 0..${n - 2} FOR doc IN ${cn} FILTER doc.value1 >= ${pad} && doc.value1 < ${pad2} RETURN doc`;
+
+      const opts = { optimizer: { rules: ["-interchange-adjacent-enumerations"] } };
+
+      let nodes = AQL_EXPLAIN(q, null, opts).plan.nodes;
+      let indexNodes = nodes.filter((node) => node.type === 'IndexNode');
+      assertEqual(1, indexNodes.length);
+      assertEqual(1, indexNodes[0].indexes.length);
+      assertEqual("UnitTestsIndex", indexNodes[0].indexes[0].name);
+      
+      let qr = db._query(q, null, opts);
+      let stats = qr.getExtra().stats;
+      assertEqual(1, stats.cursorsCreated);
+      assertEqual(n - 2, stats.cursorsRearmed);
+      let results = qr.toArray();
+      assertEqual(n - 1, results.length);
+
+      for (let i = 0; i < n - 1; ++i) {
+        let doc = results[i];
+        assertEqual(prefix + String(i).padStart(5, "0"), doc.value1);
+        assertEqual(i, doc.value2);
+      }
+    },
+    
+    testLookupByRangeOnSecondAttribute : function () {
+      const q = `FOR i IN 0..${n - 2} FOR doc IN ${cn} FILTER doc.value1 == ${pad} && doc.value2 >= i && doc.value2 < i + 1 RETURN doc`;
+
+      const opts = { optimizer: { rules: ["-interchange-adjacent-enumerations"] } };
+
+      let nodes = AQL_EXPLAIN(q, null, opts).plan.nodes;
+      let indexNodes = nodes.filter((node) => node.type === 'IndexNode');
+      assertEqual(1, indexNodes.length);
+      assertEqual(1, indexNodes[0].indexes.length);
+      assertEqual("UnitTestsIndex", indexNodes[0].indexes[0].name);
+
+      let qr = db._query(q, null, opts);
+      let stats = qr.getExtra().stats;
+      assertEqual(1, stats.cursorsCreated);
+      assertEqual(n - 2, stats.cursorsRearmed);
+      let results = qr.toArray();
+      assertEqual(n - 1, results.length);
+
+      for (let i = 0; i < n - 1; ++i) {
+        let doc = results[i];
+        assertEqual(prefix + String(i).padStart(5, "0"), doc.value1);
+        assertEqual(i, doc.value2);
+      }
+    },
+    
+    testLookupBySingleAttributeUsingIn : function () {
+      const q = `FOR i IN 0..${n - 2} FOR doc IN ${cn} FILTER doc.value1 IN [${pad}, 'piff'] RETURN doc`;
```

Comment:
How about a test that combines `IN` on `value1` with some condition on `value2`?
What about queries like `FILTER (doc.value1 >= A AND doc.value1 < B) OR (doc.value1 >= X AND doc.value1 < Y)`?

Reply:
Tests added.